File: JavaDemoGithubPractice/app/src/main/java/com/example/javademogithubpractice/common/SizedMap.java


package com.example.javademogithubpractice.common;

import java.util.LinkedHashMap;

/**
 * Created by ThirtyDegreesRay on 2017/8/22 18:46:06
 */

public class SizedMap<K,V> extends LinkedHashMap<K,V> {

    private int maxSize ;
    private final int DEFAULT_SIZE = 64;

    public SizedMap() {
        super();
        this.maxSize = DEFAULT_SIZE;
    }

    public SizedMap(int maxSize) {
        super();
        this.maxSize = maxSize;
    }

    @Override
    protected boolean removeEldestEntry(Entry<K, V> eldest) {
        return size() > maxSize;
    }

}

File: KotlinDemoGithubProject/app/src/main/java/com/example/kotlindemogithubproject/common/SizedMap.kt
package com.example.kotlindemogithubproject.common

import java.util.LinkedHashMap
import java.util.Map


class SizedMap<K, V> : LinkedHashMap<K, V> {

    private var maxSize: Int = 0
    private val DEFAULT_SIZE = 64

    constructor() : super() {
        this.maxSize = DEFAULT_SIZE
    }

    constructor(maxSize: Int) : super() {
        this.maxSize = maxSize
    }

    override fun removeEldestEntry(eldest: MutableMap.MutableEntry<K, V>?): Boolean {
        return size > maxSize
    }

}
